### PR TITLE
Adding Team Lead and Engineering Manager

### PR DESCRIPTION
**Pull request by MarkSiedle**
For people considering a Team Lead or Engineering Manager position at Octopus, we want Octopus People to be very clear about expectations. It can help managers to understand when someone may/may not be ready to make the move to another role and helps people internally understand their career progression options and again, the expectations.

We already have a `Leadership` document outlining (in great detail) the expectations of a leadership role.

This PR builds on that by referencing the existing `Leadership` document and building on it with traits that are unique to the technical requirements of an Engineering Team Lead and Engineering Manager, and pulls some leadership qualities from the `Solution Architect` role description (since these are now a separate band).

### CODE REVIEW

**Comment by Waldo000000** [comment]
To summarize some feedback from this morning:
- [ ] Tweak SSE to replace mentions of shipping `full products` to more like `shaped pitches`
- [ ] Make sure that SA still sufficiently emphasizes the need to "bring people along with you" in the process of pitching, convincing, and making sure others understand the solution throughout the process (raised by @pawelpabich)

File: Engineering.md [hunk 1]
```diff
@@ -79,47 +93,43 @@
 - I am actively introspective and take time to work on my personal and professional development.
 - I happily take responsibility for my own mistakes and I learn from the mistakes of others.
 
+## Team Lead
+
+> I am trusted to autonomously lead people through complex problems with a proven record of delivering ideas from inception to their successful completion.
+> A Team Lead emcompasses all of the qualities listed in [Leadership](Leadership.md) and is usually combined with either a Senior Software Engineer or Solution Architect role.
+
+### I am trusted to lead a team of diverse technical roles (from Architects, Engineers, Designers and Testers) to ship full products to production
```
Comment (Waldo000000) [inline Engineering.md]: What's an example of a "full product" here? Should this be "shaped pitch"?

Comment (MarkSiedle) [reply Engineering.md]: Yes! Missed one, ty

File: Engineering.md [hunk 2]
```diff
@@ -35,16 +35,30 @@
...
 > I am a highly sought-after member of an Octopus team with a proven record of shipping high-quality code to production.
 
-### I am trusted to ship full products (or vertical slices of larger products) from idea to production
+### I am trusted to ship shaped pitches (or vertical slices of larger products) from idea to production
 
 - I am proficient with my tools and the core building blocks used by our team.
 - I can confidently step outside my comfort zone and adapt quickly to new technologies.
 - I am proficient at qualifying and reducing technical debt.
-- I am comfortable pulling together the Lego blocks of a full product.
+- I am comfortable pulling together the Lego blocks of a shaped pitches.
```
Comment (pawelpabich) [inline Engineering.md]: `a shaped pitch` or `shaped pitches`

File: Engineering.md [hunk 3]
```diff
@@ -79,47 +93,43 @@
...
+- I am a well-rounded leader of people, known for having a collaborative leadership style.
+- I am proficient at leading software projects using agile practices.
+- I am comfortable with transparently assessing risk, making recommendations, escalating appropriately, and dealing with the consequences along the way.
+- I can confidently and charismatically pitch ideas, positively influencing my team to take decisive action based on reaching consensus.
+- I am an adept communicator, and can effectively steer technical and non-technical conversations to positive outcomes over any medium.
+
 ## Solution Architect
 
-> I am a well-rounded developer, architect, problem solver, and leader of people. I can be trusted to take a brand new idea from inception to production with a small group of people.
+> I am a well-rounded developer, architect and problem solver. I can be trusted to take a brand new idea from inception to production, bringing people along with me throughout the process and providing clarity for our team.
```
Comment (pawelpabich) [inline Engineering.md]: Nice :)

File: Engineering.md (hunk 2 above)
Comment (MarkSiedle) [reply Engineering.md]: 🤦‍♂ Good pickup, ty

File: Engineering.md [hunk 4]
```diff
@@ -79,47 +93,58 @@
...
+- I am comfortable with transparently assessing risk, making recommendations, escalating appropriately, and dealing with the consequences along the way.
+- I can confidently and charismatically pitch ideas, positively influencing my team to take decisive action based on reaching consensus.
+- I am an adept communicator, and can effectively steer technical and non-technical conversations to positive outcomes over any medium.
+
+## Engineering Manager
+
+_We consider an engineering manager to be a full time role. See the [manager's handbook](https://octopushq.atlassian.net/wiki/spaces/IN/pages/628981849/Managers+at+Octopus)._
+
+> I take responsibility for a tribe of people in the engineering team and the results they ultimately deliver.
+> An Engineering Manager emcompasses all of the qualities listed in [Leadership](Leadership.md) and is usually a full-time role.
```
Comment (Waldo000000) [inline Engineering.md]:
```suggestion
> An Engineering Manager encompasses all of the qualities listed in [Leadership](Leadership.md) and is usually a full-time role.
```

File: Engineering.md [hunk 5]
```diff
@@ -79,47 +93,58 @@
...
+
+### I am trusted to lead a team of diverse technical roles (from Architects, Engineers, Designers, and Testers) to ship shaped pitches to production
+
+- I am a well-rounded leader of people, known for having a collaborative leadership style.
+- I am proficient at leading software projects using agile practices.
+- I am comfortable with transparently assessing risk, making recommendations, escalating appropriately, and dealing with the consequences along the way.
+- I can confidently and charismatically pitch ideas, positively influencing my team to take decisive action based on reaching consensus.
+- I am an adept communicator, and can effectively steer technical and non-technical conversations to positive outcomes over any medium.
+
+## Engineering Manager
```
Comment (Waldo000000) [inline Engineering.md]: **First impression**: These characteristics form a great description of my ideal manager; love it 💙💙💙

**Second thought**: Does this description aim to answer these kinds of more specific questions?
- Do Managers at Octopus influence architecture? Write code? Are they ultimately responsible for any cross-cutting concerns like security? (etc)
- Do Managers at Octopus influence product? (see: betting table)
- Do Managers hire/fire/do salary reviews, etc?

Comment (michaelnoonan) [reply Engineering.md]: > Do Managers at Octopus influence architecture? Write code? Are they ultimately responsible for any cross-cutting concerns like security? (etc)

In our current situation, yes, but I think that is a byproduct of us historically being Solution Architects. In this section, I want to focus on the unique value proposition of an Engineering Manager, explaining their first priorities, and if they contribute extra value outside this, that could be OK.

> Do Managers at Octopus influence product? (see: betting table)

I have the same sentiment here also. Our current engineering managers are involved in the betting table more as a product of history and capability at the moment, and perhaps that will change over time? One unique value we contributed to the betting table was capacity planning: which bets can we actually achieve with the people we have in each tribe?

> Do Managers hire/fire/do salary reviews, etc?

Yes, I will definitely add something around this. This is part of where I was going with identifying and dealing with dysfunction.

File: Engineering.md [hunk 6]
```diff
@@ -79,47 +93,60 @@
...
```
Comment (MarkSiedle) [inline Engineering.md]: Love it, and it complements and aligns well with the manager's handbood 👍 Nicely done.

File: Engineering.md [hunk 7]
```diff
@@ -35,16 +35,30 @@
 - I go out of my way to develop a friendly and professional rapport with my customers and their team members.
 - My contributions to team discussions are net-positive, informative, and helpful.
 
+## Quality Assurance Engineer
```
Comment (michaelnoonan) [inline Engineering.md]: I moved QAE as-is into this location.

**Pull request merged**